packages/cloudflare/src/cli/templates/worker.ts

```diff
@@ -11,53 +11,61 @@ import { createALSProxy } from "./utils";
 
 const NON_BODY_RESPONSES = new Set([101, 204, 205, 304]);
 
+const processEnvALS = new AsyncLocalStorage<Record<string, unknown>>();
 const cloudflareContextALS = new AsyncLocalStorage<CloudflareContext>();
 
 // Note: this symbol needs to be kept in sync with the one defined in `src/api/get-cloudflare-context.ts`
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 (globalThis as any)[Symbol.for("__cloudflare-context__")] = createALSProxy(cloudflareContextALS);
 
+globalThis.process = {
+  ...globalThis.process,
+  // @ts-expect-error - populated when we run inside the ALS context
+  env: createALSProxy(processEnvALS),
```

Comment:
@petebacondarwin @dario-piotrowicz is that ok? I think we do not want to replace the env, right?

Reply:
I think this is valid, what we currently do is:
```ts
globalThis.process.env = { ...globalThis.process.env, ...env };
```

which just drops everything onto `globalThis.process` potentially causing leaks, so I think that using ALS instead is the right approach

Although I'd argue, why using ALS only for `env` and not the whole `process` object?

(As we do [in next-on-pages](https://github.com/cloudflare/next-on-pages/blob/96c331d20e679ebad0e43153430786685dbbcdec/packages/next-on-pages/src/buildApplication/generateGlobalJs.ts#L20-L30))

Reply:
Happy to change it if you'd prefer it for for the entire process object, I don't mind either way

Reply:
Honestly I don't think that it will make a significant difference...

But on the other side of things, it's also a trivial change to make and makes things potentially more robust without any downside no?

Do you think there's any benefit in just doing this for the `env`?

Reply:
i dont think there's any benefit to one over the other, hence why i dont mind whichever you'd prefer :)

Reply:
I feel like there's a veeeery veeeery slight benefit in wrapping the process object entirely, so yeah I would prefer going with that if you'd be ok with it 🙂

Reply:
Actually it will cause a problem. If we do it on process instead of process.env, some apps where things may try to access parts of process.env which Cloudflare decides to implement, but we would have no way of making that available to them since we spread globalthis.process because, well, that is a proxy now.

Reply:
Ah ok, sorry my bad 🙇 , thanks for giving it a try 🙂👍